draco/analysis/powerspec.py

```diff
@@ -0,0 +1,1504 @@
+
+    # The gridded u and v coordinates
+    u = kperp_to_u(k_x, redshift)
+    v = kperp_to_u(k_y, redshift)
+
+    # Estimate the line-of-sight Fourier modes
+    k_para = delays_to_kpara(delays, redshift)
+
+    return k_x, k_y, u, v, k_para
+
+
+def image_to_uv(data, ra, dec, window="tukey-0.5"):
+    """Spatial FFT along RA and DEC axes of the data cube.
+
+    Parameters
+    ----------
+    data : np.ndarray[ra,el]
+       The data, whose spatial FFT will be computed along RA and Dec axes.
+    ra : np.array(ra)
+      RA of the map in degrees.
+    dec : np.array(dec)
+      Dec of the map in degrees.
+    window: window available in :func:`draco.util.tools.window_generalised()`, optional
+       Apply an apodisation function. Default: 'tukey-0.5'.
+
+    Returns
+    -------
+      data_cube : np.ndarray[kx,ky]
+         The 2D spatial FFT of the data cube in (kx,ky) or (u,v) domain.
+    """
+    # Find the Fourier norm for the FFT
+    # The norm is mentioned here - https://numpy.org/doc/2.0/reference/routines.fft.html
+    FT_norm = 1 / float(np.prod(np.array(data.shape)))
+
+    if window:
+        x_ra = (ra - ra[0]) / (ra[-1] - ra[0])
+        x_dec = (dec - dec[0]) / (dec[-1] - dec[0])
+        w_ra = tools.window_generalised(x_ra, window=window)
+        w_dec = tools.window_generalised(x_dec, window=window)
+
+        # estimate the equivalent noise bandwidth for the tapering window
+        NEB_ra = noise_equivalent_bandwidth(ra.size, window)
+        NEB_dec = noise_equivalent_bandwidth(dec.size, window)
+        taper_window = np.outer(w_ra[:, np.newaxis], w_dec[np.newaxis, :])
+        data *= taper_window
+        uv_map = np.fft.fftshift(np.fft.fft2(data))
+
+    else:
+        uv_map = np.fft.fftshift(np.fft.fft2(data))
+        NEB_ra = NEB_dec = 1.0
+
+    return uv_map * FT_norm, NEB_ra, NEB_dec
+
+
+def vol_normalization(ra, dec, freq, redshift, cosmo=None):
+    """Estimate the volume normalization factor for the power spectrum.
+
+    Parameters
+    ----------
+    ra : np.array[ra]
+      The RA array in deg unit.
+    dec : np.narray[dec]
+      The DEC array in deg unit.
+    freq : np.array[freq]
+      The freq array in MHz.
+    redshift : float
+      redshift at the centre of the band.
+    cosmo: Cosmology object
+        Default is cora.util.Cosmology() default.
+
+    Returns
+    -------
+    norm : float
+      The  Ppower spectrum normalization factor in Mpc^3 unit
+    """
+    if cosmo is None:
+        cosmo = get_cosmo()
+
+    nra = ra.size
+    ndec = dec.size
+    nfreq = freq.size
+
+    # Mean resolution in RA and DEC and convert that to radian
+    res_ra_radian = np.deg2rad(np.mean(np.diff(ra)))
+    res_dec_radian = np.deg2rad(np.mean(np.diff(dec)))
+
+    # Comoving distance
+    DMz = dRperp_dtheta(redshift, cosmo=cosmo)  # in [h^-1 Mpc]
+
+    # Convert the RA and DEC resolution to Mpc unit
+    dx_Mpc = DMz * res_ra_radian
```

Comment:
```suggestion
    dx_Mpc = DMz * res_ra_radian  * np.mean(np.cos(np.deg2rad(dec)))
```
As above, I think you need a factor of cos(dec) here.